src/irsForms/ScheduleEIC.ts

```diff
@@ -1,11 +1,349 @@
+  f8814?: F8814
+  pub596Worksheet1: Pub596Worksheet1
+  qualifyingCutoffYear: number = 1996
+  investmentIncomeLimit: number = 3650
+
+  constructor (tp: TP) {
+    this.tp = new TaxPayer(tp)
+    this.f2555 = new F2555(tp)
+    this.f4797 = new F4797(tp)
+    this.f8814 = new F8814(tp)
+    this.pub596Worksheet1 = new Pub596Worksheet1(tp)
+  }
+
+  // At least key children, and max [NonMFJLimit, MFJLimit] income
+  step1ChildrenIncomes: Array<[number, number]> = [
+  // Nonmfj, mfjlimit
+    [15820, 21710], // 0 dependents
+    [41756, 47646], // 1
+    [47440, 53330], // 2
+    [50954, 56844] // 3
+  ]
+
+  // instructions step 1.1
+  passIncomeLimit = (f1040: F1040): boolean => {
+    if (this.tp.tp.filingStatus !== undefined) {
+      const incomeLimits = federal.EIC.caps[this.tp.tp.filingStatus]
+      if (incomeLimits !== undefined) {
+        const limit = incomeLimits[Math.min(this.qualifyingDependents.length, incomeLimits.length - 1)]
+        return (f1040.l11() ?? 0) < limit
+      }
+    }
+    return false
+  }
+
+  // Step 1.2, todo, boths spouses must have a SSN issued before 2020 due date
+  // and without work restriction and valid for eic purposes
+  validSSNs = (): boolean => {
+    unimplemented('Step 1.2 (valid SSNs) unchecked')
+    return true
+  }
+
+  // Step 1.3
+  allowedFilingStatus = (): boolean => this.tp.tp.filingStatus !== FilingStatus.MFS
+
+  // Step 1.4
+  allowedFilling2555 = (): boolean => !precludesEIC(checks2555)(this.f2555)
+
+  // Step 1.5 nonResidentAlien
+  allowedNonresidentAlien = (): boolean => {
+    unimplemented('Step 1.5, Not checking non-resident alien')
+    return true
+  }
+
+  // step 2, question 1
+  investmentIncome = (f1040: F1040): number => sumFields([
+    f1040.l2a(),
+    f1040.l2b(),
+    f1040.l3b(),
+    Math.max(f1040.l7() ?? 0, 0)
+  ])
+
+  passInvestmentIncomeLimit = (f1040: F1040): boolean =>
+    this.investmentIncome(f1040) < federal.EIC.maxInvestmentIncome
+
+  // Todo, step 2, question 3
+  f4797AllowsEIC = (): boolean => !precludesEIC(checks4797)(this.f4797)
+
+  // Todo, instruction 2.4.1
+  filingScheduleE = (): boolean => {
+    unimplemented('Not checking Schedule E')
+    return false
+  }
+
+  // 2.4.2
+  passIncomeFromPersonalProperty = (): boolean => {
+    unimplemented('Not checking personal property income')
+    return true
+  }
+
+  // 2.4.3
+  passForm8814 = (): boolean => !precludesEIC(checks8814)(this.f8814)
+
+  // 2.4.4
+  incomeOrLossFromPassiveActivity = (): boolean => {
+    unimplemented('Not checking passive activity')
+    return false
+  }
+
+  // 2.4.5
+  passPub596 = (): boolean => precludesEIC(checksPub596)(this.pub596Worksheet1)
+
+  // 3.1
+  atLeastOneChild = (): boolean => this.qualifyingDependents().length > 0
+
+  // 3.2, 4.4
+  jointReturn = (): boolean => this.tp.tp.filingStatus === FilingStatus.MFJ
+
+  // 3.3, 4.5
+  qualifyingChildOfAnother = (): boolean => {
+    unimplemented('3.3: Not checking qualifying child of another')
```

Comment:
I'd missed this in my first skim. Looks like the questions related to divorce will be implemented later.